src/engine/Server.cpp

```diff
@@ -20,22 +20,22 @@ template <typename T>
 using Awaitable = Server::Awaitable<T>;
 
 // __________________________________________________________________________
-Server::Server(const int port, const int numThreads, size_t maxMemGB,
-               std::string accessToken)
+Server::Server(unsigned short port, int numThreads, size_t maxMemGB,
+               std::string&& accessToken, bool usePatternTrick)
     : _numThreads(numThreads),
       _port(port),
-      accessToken_(accessToken),
-      _allocator{ad_utility::makeAllocationMemoryLeftThreadsafeObject(
-                     maxMemGB * (1ull << 30u)),
-                 [this](size_t numBytesToAllocate) {
-                   _cache.makeRoomAsMuchAsPossible(
-                       static_cast<double>(numBytesToAllocate) / sizeof(Id) *
-                       MAKE_ROOM_SLACK_FACTOR);
-                 }},
+      accessToken_(std::move(accessToken)),
```

Comment:
For the access token I would suggest pass-by-value and then move.
Your code forces to move the access token in, I am not sure if that is not too aggressive here. But we can talk about the philosophy here (there is some C++23 about this).